reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -0,0 +1,677 @@
+
+        Flux<String> switchTransformed = publisher.doOnRequest(requested::addAndGet)
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux
+                                                          .map(String::valueOf)
+                                                          .filter(e -> false));
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+
+        Assert.assertEquals(10001L, requested.get());
+    }
+
+    @Test
+    public void shouldErrorOnOverflowTest() {
+        TestPublisher<Long> publisher = TestPublisher.createCold();
+
+        Flux<String> switchTransformed = publisher.flux()
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf));
+
+        publisher.next(1L);
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectNext("1")
+                    .then(() -> publisher.next(2L))
+                    .expectError()
+                    .verify(Duration.ofSeconds(10));
+
+        publisher.assertWasRequested();
+        publisher.assertNoRequestOverflow();
+    }
+
+    @Test
+    public void shouldPropagateonCompleteCorrectly() {
+        Flux<String> switchTransformed = Flux.empty()
+                                             .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf));
+
+        StepVerifier.create(switchTransformed)
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+    }
+
+    @Test
+    public void shouldPropagateOnCompleteWithMergedElementsCorrectly() {
+        Flux<String> switchTransformed = Flux.empty()
+                                             .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf)
+                                                                                           .mergeWith(Flux.just("1", "2", "3")));
+
+        StepVerifier.create(switchTransformed)
+                    .expectNext("1", "2", "3")
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+    }
+
+    @Test
+    public void shouldPropagateErrorCorrectly() {
+        Flux<String> switchTransformed = Flux.error(new RuntimeException("hello"))
+                                             .transform(flux -> new FluxSwitchOnFirst<>(
+                                                     flux,
+                                                     (first, innerFlux) -> innerFlux.map(
+                                                             String::valueOf)));
+
+        StepVerifier.create(switchTransformed)
+                    .expectErrorMessage("hello")
+                    .verify(Duration.ofSeconds(10));
+    }
+
+    @Test
+    public void shouldBeAbleToBeCancelledProperly() {
```

Comment:
one good alternative version of testing this would be by using eg. `take(1)`. maybe add that as a second test `shouldBeableToBeCancelledProperly2`?

Reply:
done